sql/analyzer/indexes.go

```diff
@@ -72,13 +72,13 @@ func getIndexes(
 			if rightIdx, ok := rightIndexes[table]; ok {
 				if canMergeIndexes(leftIdx.lookup, rightIdx.lookup) {
 					var allRanges sql.RangeCollection
-					allRanges = append(sql.RangeCollection{}, leftIdx.lookup.Ranges()...)
-					allRanges = append(allRanges, rightIdx.lookup.Ranges()...)
+					allRanges = append(sql.RangeCollection{}, leftIdx.lookup.Ranges...)
+					allRanges = append(allRanges, rightIdx.lookup.Ranges...)
 					newRanges, err := sql.RemoveOverlappingRanges(allRanges...)
 					if err != nil {
 						return nil, nil
 					}
-					newLookup, err := leftIdx.lookup.Index().NewLookup(ctx, newRanges...)
```

Comment:
This is a bug, I'm pretty sure: this could return nil for the newLookup before if the index couldn't support that range scan. We need something similar here.

Reply:
I added checks to the static lookup points, where we have ranges for checks. The other cases should all be point lookups. If your index does not support point lookups then that seems like a separate issue.